pilot/pkg/networking/plugin/authz/rbac.go

```diff
@@ -224,6 +254,27 @@ func createDynamicMetadataMatcher(k, v string) *metadata.MetadataMatcher {
 			},
 		}
 	}
+	return stringMatcher
+}
+
+// createDynamicMetadataMatcher creates a MetadataMatcher for the given key, value pair.
+func createDynamicMetadataMatcher(k, v string) *metadata.MetadataMatcher {
+	keys, err := generateKeys(k)
+	if err != nil {
+		return nil
+	}
+	forceRegexPattern := false
+	if k == attrSrcNamespace {
+		// Change the value to a regular expression to match the namespace part.
+		v = fmt.Sprintf(`*/ns/%s/*`, v)
+		forceRegexPattern = true
+	}
+
+	if len(keys) == 2 && keys[0] == attrRequestClaims && keys[1] == groupsClaim {
```

Comment:
Why don't you have condition on k instead. This condition is over complicated.

Reply:
Done.